spec/models/materialized_patient_summary_spec.rb

```diff
@@ -252,18 +252,90 @@ def refresh_view
   end
 
   describe "#ltfu?" do
-    it "is true if patient was registered over a year ago w/o any BPs recorded" do
-      ltfu_patient = create(:patient, recorded_at: 2.years.ago)
+    it "is true if patient was registered over a year ago w/o any BPs, BSs, PDs or appointments recorded" do
+      ltfu_patient = create(:patient, recorded_at: 365.days.ago)
       refresh_view
 
       expect(described_class.find(ltfu_patient.id).ltfu?).to eq true
     end
 
     it "is false if patient registerd w/i the LTFU time" do
-      not_ltfu_patient = create(:patient, recorded_at: 1.month.ago)
+      not_ltfu_patient = create(:patient, recorded_at: 364.days.ago)
       refresh_view
 
       expect(described_class.find(not_ltfu_patient.id).ltfu?).to eq false
     end
+
+    context "patient records a BP" do
+      it "is false if patient recorded a BP w/i the LTFU time" do
+        not_ltfu_patient = create(:patient, recorded_at: 365.days.ago)
+        create(:blood_pressure, patient: not_ltfu_patient, device_created_at: 364.days.ago)
+        refresh_view
+
+        expect(described_class.find(not_ltfu_patient.id).ltfu?).to eq false
+      end
+
+      it "is true if patient recorded a BP not w/i the LTFU time" do
+        ltfu_patient = create(:patient, recorded_at: 365.days.ago)
+        create(:blood_pressure, patient: ltfu_patient, device_created_at: 365.days.ago)
+        refresh_view
+
+        expect(described_class.find(ltfu_patient.id).ltfu?).to eq true
+      end
+    end
+
+    context "patient records a BS" do
+      it "is false if patient recorded a BS w/i the LTFU time" do
+        not_ltfu_patient = create(:patient, recorded_at: 365.days.ago)
+        create(:blood_sugar, patient: not_ltfu_patient, device_created_at: 364.days.ago)
```

Comment:
Ah I see, the blood sugar factory automatically sets the `recorded_at` to match the `device_created_at`. Let's set the `recorded_at` explicitly here anyway, since that makes it much more evident how our underlying query is implemented.

Reply:
Ah I see, the blood sugar factory automatically sets the `recorded_at` to match the `device_created_at`. Let's set the `recorded_at` explicitly here anyway, since that makes it much more evident how our underlying query is implemented.

Reply:
Right, let's use `recorded_at` instead